### PR TITLE
Add scripts for meeting Rival and Token

### PR DESCRIPTION
**Pull request by TonyArra**
This PR adds three new events to the beginning of the game:

- Meeting Rival in your bedroom
- Meeting Team Token Grunt (still using old sprites)
- Rival follows you back to professor's lab

It also adds the `/includes/garticmon.s` file which we can use to store any new flags/vars that we use in our scripts.

A chunk of the Team Token intro was copied from XSE for Professor Oak's movement, and converted into CFRU.

### CODE REVIEW

File: strings/scripts/pallet_town/player_room.string [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#org @gText_PlayerRoom_Rival_WakeUp
+[GREEN][PLAYER]...\p[PLAYER]...\pWake up, [PLAYER]...!!
+
+#org @gText_PlayerRoom_Rival_Late
+[GREEN][RIVAL]: Come on come on!\p Or we're gonna be late!
+
+#org @gText_PlayerRoom_Player_Festival1
+[RED][PLAYER]: Late...?\pOh, right\pThe Millennial Art Festival is today.
```
Comment (Stevebel) [inline strings/scripts/pallet_town/player_room.string]: Like the Green for Rival but Fire Red uses red text for female characters unless we change that

File: assembly/overworld_scripts/pallet_town/pallet_token_intro.s [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+  .byte end_m
+
+FollowProfessor_Tile0:
+  applymovement PERSON_PROFESSOR Move_Professor_FollowProfessor_Tile0
+  applymovement PLAYER Move_Player_FollowProfessor_Tile0
+  waitmovement 0x0
+  return
+
+Move_Professor_FollowProfessor_Tile0:
+  .byte 0x10 @Step Down (Normal)
```
Comment (Stevebel) [inline assembly/overworld_scripts/pallet_town/pallet_token_intro.s]: Can we replace these Hex value + comment with the kind of constants used by the grunt?

File: strings/scripts/pallet_town/pallet_token_intro.string [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#org @gText_PalletTokenIntro_Grunt_HeyKids
+Psst...\pHey kids...\pYa ever heard of NFTs?
```
Comment (Stevebel) [inline strings/scripts/pallet_town/pallet_token_intro.string:2]: Check it... only the finest celebrity-endorsed NFTs

**Review by Stevebel — CHANGES_REQUESTED** [review]

File: strings/scripts/pallet_town/player_room.string (hunk 1 above)
Comment (TonyArra) [reply strings/scripts/pallet_town/player_room.string]: @Stevebel Changed to orange until we figure out somethin better

File: assembly/overworld_scripts/pallet_town/pallet_token_intro.s (hunk 2 above)
Comment (TonyArra) [reply assembly/overworld_scripts/pallet_town/pallet_token_intro.s]: @Stevebel fixed

**Pull request merged**